shell/platform/android/io/flutter/embedding/android/FlutterActivity.java

```diff
@@ -28,62 +29,141 @@
 import io.flutter.view.FlutterMain;
 
 /**
- * {@code Activity} which displays a {@link io.flutter.view.FlutterView} that takes up all
- * available space.
+ * {@code Activity} which displays a {@link FlutterFragment} that takes up all available space.
  *
  * {@link FlutterActivity} is the simplest and most direct way to integrate Flutter within an
  * Android app.
- * TODO(mattcarroll): how exactly does FlutterActivity need to be leveraged to display Flutter content?
  *
- * If Flutter is needed in a location that cannot use an {@code Activity>}, consider using
+ * By default, {@link FlutterActivity} configures itself to do the following:
+ *  - no splash screen
+ *  - asks {@link FlutterMain#findAppBundlePath(Context)} for the path to the Dart app bundle
+ *  - uses a default Dart entrypoint of "main"
+ *  - uses a default initial route of "/" within the Flutter app
+ *
+ * The display of a splash screen, app bundle path, Dart entrypoint, and initial route can each be
+ * controlled by overriding their respective methods in a subclass:
+ *  - {@link #isSplashScreenDesired()}
+ *  - {@link #getAppBundlePath()}
+ *  - {@link #getDartEntrypoint()}
+ *  - {@link #getInitialRoute()}
+ *
+ * If Flutter is needed in a location that cannot use an {@code Activity}, consider using
  * a {@link FlutterFragment}. Using a {@link FlutterFragment} requires forwarding some calls from
  * an {@code Activity} to the {@link FlutterFragment}.
  *
  * If Flutter is needed in a location that can only use a {@code View}, consider using a
- * {@link io.flutter.view.FlutterView}. Using a {@link io.flutter.view.FlutterView} requires
- * forwarding some calls from an {@code Activity}, as well as forwarding lifecycle calls from
- * an {@code Activity} or a {@code Fragment}.
+ * {@link FlutterView}. Using a {@link FlutterView} requires forwarding some calls from an
+ * {@code Activity}, as well as forwarding lifecycle calls from an {@code Activity} or a
+ * {@code Fragment}.
  *
- * @see FlutterFragment
- *
- * @see io.flutter.view.FlutterView
+ * @see FlutterFragment, which presents a Flutter app within a {@code Fragment}
+ * @see FlutterView, which renders the UI for a {@link FlutterEngine}
  */
 @SuppressLint("Registered")
 @TargetApi(Build.VERSION_CODES.ICE_CREAM_SANDWICH_MR1)
-public class FlutterActivity extends Activity {
+public class FlutterActivity extends FragmentActivity {
   private static final String TAG = "FlutterActivity";
 
-  private static final String EXTRA_SHOW_SPLASH_SCREEN = "show_launch_screen";
+  // Meta-data arguments
   // TODO: where did this package path come from? is this what it should be?
```

Comment:
nit: Here and elsewhere `TODO(username)` as in [style guide](https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo#avoid-checking-in-comments-that-ask-questions) example.